vignettes/Intro_PoolDilutionR.Rmd

```diff
@@ -0,0 +1,368 @@
+             m_prec = 0.001, # instrument precision for total pool size, as standard deviation
+             ap_prec = 1, # instrument precision for atom percent, as standard deviation
+)
+```
+
+Notice that `pdr_optimize` calculated or looked up three parameters that we didn't provide:
+`frac_P`, the fractionation rate of production; `frac_k`, the fractionation rate of
+consumption; and `k0`, the initial value for the first order rate constant for consumption.
+The first two are taken from default values for methane as provided in the 
+package's `pdr_fractionation` dataset, while the latter is estimated from the data, using the slope of the pool size of heavy isotope over time (`n_slope = -11.635`).
+
+`pdr_optimize` returns a list with the following entries:
+
+* `par` - the final optimized values of (in this case) production (`P`) and consumption (`k`), as calculated by `optim()`
+* `value`, `counts`, `convergence`, `message` - these are all returned by `optim()`
+* `initial_par` - the initial parameters used for the model
+* `initial_other` - other settings passed to `optim()`; these typically include parameter bounds and optimization method
+
+This list has a great deal of detail, but we can also get a summarized form in
+a convenient data frame output format:
+
+```{r OneSample Results}
+pdr_optimize_tidy(time = OneSample_dat$time_days,
+                  m = OneSample_dat$cal12CH4ml + OneSample_dat$cal13CH4ml,
+                  n = OneSample_dat$cal13CH4ml,
+                  P = 0.1,
+                  pool = "CH4",
+                  m_prec = 0.001,
+                  ap_prec = 1)
+```
+
+From `k`, the gross consumption rate can be calculated for any pool size of interest.  
+
+## Multi-sample Processing
+  
+To demonstrate additional capabilities, we use the entire `Morris2023` dataset:  
+  
+```{r Multi-sample Example Input, fig.height = 5, fig.width = 8, fig.align='center'}
+# create long data for plotting
+library(tidyr)
+long_Morris2023 <- pivot_longer(Morris2023, cols = cal12CH4ml:AP_obs)
+
+library(ggplot2)
+ggplot(data = long_Morris2023,
+       aes(time_days, value, color = id)) +
+  geom_point() + geom_line() +
+  facet_wrap(~name, scales = "free_y")
+```
+
+
+### Visual Summary of Multisample Output
+
+Now we show a graphical summary of the results from running these samples through `pdr_optimize()` with the same arguments as shown for single sample above.  
+ 
+```{r lapply(PoolDilutionR)}
+samples <- split(Morris2023, Morris2023$id)
+#lapply(samples, pdr_optimize_tidy)
+all_results <- lapply(samples, FUN = function(samples) {
+   pdr_optimize_tidy(time = samples$time_days,
+                     m = samples$cal12CH4ml + samples$cal13CH4ml,
+                     n = samples$cal13CH4ml,
+                     P = 0.1,
+                     pool = "CH4",
+                     m_prec = 0.001,
+                     ap_prec = 1,
+                     include_progress = FALSE)}
+   )
+all_results <- do.call(rbind, all_results)
```

Comment:
Here I bring together the results from lapply, the sample id's are carried through, but it's a little clunky. Ideas for improving the tidy output? @bpbond

Reply:
I don't have a cleaner way to do it without using `dplyr`.